03-task-manager/starter/controllers/tasks.js

```diff
@@ -1,29 +1,83 @@
 const Task = require("../models/Task");
-const getAllTasks = (req, res) => {
-  res.send("get all tasks");
-};
+const asyncWrapper = require('../middleware/async')
+const {
+	createCustomError
+} = require('../errors/custom-error')
 
-const createTask = async (req, res) => {
-  const task = await Task.create(req.body);
-  res.status(201).json({ task });
-};
 
-const getTask = (req, res) => {
-  res.json({ id: req.params.id });
-};
+const getAllTasks = asyncWrapper(async (req, res) => {
+	const tasks = await Task.find({})
+	res.status(200).json({
+		tasks
+	})
+});
 
-const updateTask = (req, res) => {
-  res.send("update task");
-};
+const createTask = asyncWrapper(async (req, res) => {
+	const task = await Task.create(req.body);
+	res.status(201).json({
+		task
+	});
+});
+
+const getTask = asyncWrapper(async (req, res, next) => {
+	const {
+		id: taskID
+	} = req.params
+	const task = await Task.findOne({
+		_id: taskID
+	});
+	if (!task) {
+		return next(createCustomError(`No task with id : ${taskID}`, 404))
+
+	}
+	res.status(200).json({
+		task
+	});
+});
+
+
+const deleteTask = asyncWrapper(async (req, res) => {
+	const {
+		id: taskID
+	} = req.params;
+	const task = await Task.findOneAndDelete({
+		_id: taskID
+	})
+
+	if (!task) {
+		return next(createCustomError(`No task with id : ${taskID}`, 404))
+	}
+	res.status(200).json({
+		task
+	})
+});
+
+const updateTask = asyncWrapper(async (req, res) => {
+	const {
+		id: taskID
+	} = req.params;
+
+	const task = await Task.findOneAndUpdate({
+		_id: taskID
+	}, req.body, {
+		new: true,
+		runValidators: true,
+	})
+
+	if (!task) {
+		return next(createCustomError(`No task with id : ${taskID}`, 404))
+	}
+
+
+
+});
```

Comment:
This function is currently missing a `res` response value for when it _does_ find and update a task successfully.

Currently if I try to call this endpoint Postman, it just hangs forever. Same if I try to use the UI to edit a task. Then if I refresh I can see that it was successfully updated, indicating that the call to Mongo succeeded.

With Node+Express, every controller function must either return `res` or `next`, or it must have a `res` call as it's last line.

Reply:
Hello Akos, I have edited this part of the code. Thank you

Reply:
Looks good; feel free to merge (sorry for the delay in following up)!